python/cudf/cudf/core/dataframe.py

```diff
@@ -5126,26 +5126,35 @@ def _reduce(
         if level is not None:
             raise NotImplementedError("level parameter is not implemented yet")
 
-        if numeric_only not in (None, True):
-            raise NotImplementedError(
-                "numeric_only parameter is not implemented yet"
+        source = self
+        if numeric_only:
+            numeric_cols = (
+                name
+                for name in self._data.names
+                if is_numeric_dtype(self._data[name])
             )
-        axis = self._get_axis_from_axis_arg(axis)
+            source = self._get_columns_by_label(numeric_cols)
+            if source.empty:
+                return source.astype("float64")
```

Comment:
Can we add a test to verify that we're matching pandas behavior for dtypes in the empty case?

Reply:
Ok, had to change the implementation. Hopefully the solution works for you.

Reply:
Yup manually constructing a StringIndex is a little ugly but I'm fine with it since it's an edge case and matches pandas.